src/equisolve/nn/__init__.py

```diff
@@ -0,0 +1,13 @@
+try:
+    import torch  # noqa: F401
+
+    HAS_TORCH = True
+except ImportError:
+    HAS_TORCH = False
+
+if HAS_TORCH:
+    from .module_tensor import LinearTensorMap, ModuleTensorMap  # noqa: F401
```

Comment:
naming wise, I would call these `Linear` and `Module`, and users can then use `torch.nn.Linear/torch.nn.Module` or `equisolve.nn.Linear/equisolve.nn.Module` for the different behavior.